src/scripts/components/VideoForm.js

```diff
@@ -26,14 +25,20 @@ class VideoForm extends Component<Props, Object> {
 
   constructor (props: Props) {
     super(props)
-    this.state = {
-      title: this.props.selectedVideo.title,
-      description: this.props.selectedVideo.description
-    }
+    this.state = new VideoRecord(this.props.selectedVideo)
     this.handleInputChange = this.handleInputChange.bind(this)
     this.handleSubmit = this.handleSubmit.bind(this)
   }
 
+  componentWillReceiveProps (nextProps: Props): void {
+    this.setState(nextProps.selectedVideo)
+    this.setState({
+      id: nextProps.selectedVideo.id,
+      title: nextProps.selectedVideo.title,
+      description: nextProps.selectedVideo.description
+    })
+  }
```

Comment:
This should not be necessary. If we already have all of the data in `props`, there is no need to store it in state. We should just rely on grabbing this information from `props`.

Reply:
Also, two `setState` operations in a row doesn't seem necessary, it could be combined into one (but I still recommend zero here)

Reply:
seems reasonable to me